### DISCUSSION TITLE
Does the log output support either logrus or zap?

### DISCUSSION THREAD

**Discussion by linany** [wind-c/comqtt#89] [General]
Due to constraints imposed by existing projects, log collection is currently limited to harvesting from these two specific libraries.

**Comment by wind-c** [comment]
Comqtt does not implement multiple log adaptation schemes.
This is because comqtt is a standalone broker，not as a third-party library.
But the log class is relatively independent and you can modify it.

**Reply by linany** [reply]
I got it, thanks.
